### DISCUSSION TITLE
Multi-line prompt

### DISCUSSION THREAD

**Discussion by zzhaolei** [wezterm/wezterm#1400] [Q&A] [answered]
I used `starship` and set a two-line prompt, but when I used `Scrollbackandviewport` to clear the buffer, there was only one line left.

config:
```
{
        key = "k",
        mods = "SUPER",
        action = wezterm.action {
            ClearScrollback = "ScrollbackAndViewport"
        }
}
```

![image](https://user-images.githubusercontent.com/17561793/145973206-ddfee16d-1ef4-42ef-963b-bccac9da7764.png)

**Comment by wez** [comment, marked as answer]
This sounds OK to me; until the shell redraws its prompt, how is anything to know that there should be two lines in the prompt?
Usually you can press CTRL-L to force a line editor to redraw, or perhaps just start typing text.

You may want to consider using:

```lua
{
        key = "k",
        mods = "SUPER",
        action = {
            Multiple={
                wezterm.action {
                    ClearScrollback = "ScrollbackAndViewport"
                },
                SendKey={
                    key="l",
                    mods="CTRL",
                }
             }
         }
}
```

to automate that.

See: 
https://wezfurlong.org/wezterm/config/lua/keyassignment/Multiple.html
https://wezfurlong.org/wezterm/config/lua/keyassignment/SendKey.html

**Reply by zzhaolei** [reply]
> how is anything to know that there should be two lines in the prompt?

I Can’t answer that. I didn’t know that

This configuration work:
```
{
    key = "k",
    mods = "SUPER",
    action = {
        Multiple = {
            wezterm.action {
                ClearScrollback = "ScrollbackAndViewport"
            },
            {SendKey = {key = "l", mods = "CTRL"}}
        }
    }
}
```
